sway-lib-std/src/storage/storage_vec.sw

```diff
@@ -865,14 +890,49 @@ impl<V> StorageKey<StorageVec<V>> {
             0 => Vec::new(),
             len => {
                 // Get the number of storage slots needed based on the size.
-                let bytes = len * __size_of::<V>();
+                let size_V_bytes = __size_of::<V>();
+                let bytes = match size_V_bytes < 8 {
+                    true => {
+                        // Len * size_of_word
+                        len * 8
+                    },
+                    false => {
+                        len * size_V_bytes
+                    }
+                };
                 let number_of_slots = (bytes + 31) >> 5;
                 let ptr = alloc_bytes(number_of_slots * 32);
                 // Load the stored slice into the pointer.
                 let _ = __state_load_quad(sha256(self.field_id()), ptr, number_of_slots);
-                Vec::from(asm(ptr: (ptr, bytes)) {
-                    ptr: raw_slice
-                })
+
+                match size_V_bytes < 8 {
+                    true => {
+                        let len_bytes = len * size_V_bytes;
+                        let new_vec = alloc_bytes(len_bytes);
+                        let mut i = 0;
+                        while i < len {
+                            // The stored vec is offset with 1 word per element, remove the padding for elements less than the size of a word
+                            // (size_of_word * element)
+                            ptr
+                                .add_uint_offset((8 * i))
+                                .copy_bytes_to(new_vec.add::<V>(i), size_V_bytes);
+                            i += 1;
+                        }
+
+                        Vec::from(
+                            asm(ptr: (new_vec, len_bytes)) {
+                                ptr: raw_slice
+                            },
+                        )
+                    },
+                    false => {
+                        Vec::from(
+                            asm(ptr: (ptr, bytes)) {
+                                ptr: raw_slice
+                            },
+                        )
+                    }
```

Comment:
Here too.

Reply:
In this case an `if` statement is more appropriate 👍